src/System.Drawing.Common/src/System/Drawing/Printing/PrintingServices.Unix.cs

```diff
@@ -71,20 +72,35 @@ static PrintingServices()
 
         #region Properties
 
+        /// <summary>
+        /// Returns a list of strings representing printers installed in CUPS
+        /// </summary>
         internal static PrinterSettings.StringCollection InstalledPrinters
         {
             get
             {
-                LoadPrinters();
-                PrinterSettings.StringCollection list = new PrinterSettings.StringCollection(Array.Empty<string>());
-                lock (installed_printers)
+                LazyInitializer.EnsureInitialized(ref installed_printers, () =>
```

Comment:
will this work? `PrintinServices` has a static constructor which initializes `installed_printers` to a new Hashtable. So then I don't think the `LazyInitializer` will execute because it is already initialized.

Reply:
That's right, I must have not committed removing it.

Reply:
I guess changing the constructor creates a merge conflict, but none of the inputs changed. I believe this should work, but first time using LazyInitializer so let me know if it doesn't make sense.

Reply:
So I turned it into this: `LazyInitializer.EnsureInitialized(ref installed_printers, () =>
                {
                    installed_printers = new Hashtable();
                    // Loads values into installed_printers
                    LoadPrinters();
                    // Creates list from installed_printers that we return
                    PrinterSettings.StringCollection list = new PrinterSettings.StringCollection(Array.Empty<string>());
                    foreach (object key in installed_printers.Keys)
                    {
                        list.Add(key.ToString());
                    }
                    // Setting installed_printers back to null to cause EnsureInitialized to be called again after these threads finish
                    // Also need this to dispose of the concurrently created hashtables
                    installed_printers = null;
                    return list;
                });`

If I understand correctly, this should work. I had to drop the installed_printers creation in the constructor, which seems to have triggered a merge conflict. 

Each thread is going to make a new Hashtable, which is a downside. I could instead create a new property to just cause re-initialization. Thinking about it, that might be the best way.